BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Internal/Tracer.cs

```diff
@@ -102,7 +117,26 @@ public void OnSpanEnd(Span span)
                 }
                 return;
             }
-            Sample(span);
+            else
+            {
+                RunOnEndCallbacks(span);
+            }
+            if (!span.WasAborted)
+            {
+                Sample(span);
+            }
+        }
+
+        public void RunOnEndCallbacks(Span span)
+        {
+            foreach (var callback in _onSpanEndCallbacks)
+            {
+                if (!callback.Invoke(span))
+                {
+                    span.Abort();
```

Comment:
Our notifier logic is to not run further callbacks if we return false from one – I think it would make sense to follow this here.